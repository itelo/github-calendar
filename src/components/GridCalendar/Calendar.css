.calendar-container {
  width: 563px;
  height: 88px;
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  justify-content: center;
}

.column {
  display: flex;
  flex-direction: column;
}

.weeks-rows {
  flex: 1;
  display: flex;
  flex-direction: row;
  position: relative;
}
