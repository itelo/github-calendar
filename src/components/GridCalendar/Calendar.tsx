import * as React from "react";
import Tooltip from "../Tooltip";
import DaysOfWeek from "./DaysOfWeek";
import "./Calendar.css";
import Months, { MonthData } from "./Months";
import Square from "./Square";
import CalendarTooltipMessage from "./CalendarTooltipMessage";

export type CalendarProps = {
  months: MonthData[];
  days: { date: Date; count: number }[][];
};

function Calendar(props: CalendarProps) {
  return (
    <div className="calendar-container">
      <DaysOfWeek />
      <div className="weeks-rows">
        <Months months={props.months} />
        {props.days.length > 0 &&
          props.days.map((_days, index) => (
            <div key={index} className="column">
              {_days.map(data => (
                <React.Fragment key={data.date.toString()}>
                  <Tooltip
                    message={
                      <CalendarTooltipMessage
                        date={data.date}
                        contribuitions={data.count}
                      />
                    }
                    position="top"
                  >
                    <Square key={data.date.toString()} count={data.count} />
                  </Tooltip>
                </React.Fragment>
              ))}
            </div>
          ))}
      </div>
    </div>
  );
}

export default Calendar;
