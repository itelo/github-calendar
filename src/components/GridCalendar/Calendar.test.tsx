import React from "react";
import { render } from "react-testing-library";
import "react-testing-library/cleanup-after-each";
import Calendar from "./Calendar";
import * as R from "ramda";
import dfns from "date-fns";
import treeDays from "../../teste_arvore_data.json";
import {
  extractMonthPositionFromData,
  mergeTreeJsonWithAllDays
} from "../../utils/calendar";
// import renderer from 'react-test-renderer';

describe("<Calendar />", () => {
  test("just snapshot", () => {
    const range = R.range(-1, 13);
    const initialDate = new Date("2016-10-20");

    const mockedAllDays = range.map(r => dfns.addDays(initialDate, r));
    const data = mergeTreeJsonWithAllDays(mockedAllDays, treeDays);
    const allDaysSplitted = R.splitEvery(7, data);

    const months = allDaysSplitted.reduce(extractMonthPositionFromData, []);

    const { container } = render(
      <Calendar days={allDaysSplitted} months={months} />
    );

    expect(container).toMatchSnapshot();
  });
});
