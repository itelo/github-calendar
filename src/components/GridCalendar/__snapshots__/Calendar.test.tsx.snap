// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> just snapshot 1`] = `
<div>
  <div
    class="calendar-container"
  >
    <div
      class="days-of-week-container"
    >
      <div
        class="day-of-week-container"
      >
        <p
          class="day-of-week-font"
          data-testid="day-0"
        />
      </div>
      <div
        class="day-of-week-container"
      >
        <p
          class="day-of-week-font"
          data-testid="day-1"
        >
          Mon
        </p>
      </div>
      <div
        class="day-of-week-container"
      >
        <p
          class="day-of-week-font"
          data-testid="day-2"
        />
      </div>
      <div
        class="day-of-week-container"
      >
        <p
          class="day-of-week-font"
          data-testid="day-3"
        >
          Wed
        </p>
      </div>
      <div
        class="day-of-week-container"
      >
        <p
          class="day-of-week-font"
          data-testid="day-4"
        />
      </div>
      <div
        class="day-of-week-container"
      >
        <p
          class="day-of-week-font"
          data-testid="day-5"
        >
          Fri
        </p>
      </div>
      <div
        class="day-of-week-container"
      >
        <p
          class="day-of-week-font"
          data-testid="day-6"
        />
      </div>
    </div>
    <div
      class="weeks-rows"
    >
      <div
        class="month-container"
        data-testid="month-0"
        style="left: 0px;"
      >
        <p
          class="month-font"
        >
          Oct
        </p>
      </div>
      <div
        class="column"
      >
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div
            class="tooltip-trigger"
            data-testid="tooltip-trigger"
          >
            <div
              class="square-container"
            >
              <div
                class="square nothing"
                data-testid="square-color"
              />
            </div>
          </div>
        </div>
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div
            class="tooltip-trigger"
            data-testid="tooltip-trigger"
          >
            <div
              class="square-container"
            >
              <div
                class="square nothing"
                data-testid="square-color"
              />
            </div>
          </div>
        </div>
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div
            class="tooltip-trigger"
            data-testid="tooltip-trigger"
          >
            <div
              class="square-container"
            >
              <div
                class="square little"
                data-testid="square-color"
              />
            </div>
          </div>
        </div>
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div
            class="tooltip-trigger"
            data-testid="tooltip-trigger"
          >
            <div
              class="square-container"
            >
              <div
                class="square nothing"
                data-testid="square-color"
              />
            </div>
          </div>
        </div>
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div
            class="tooltip-trigger"
            data-testid="tooltip-trigger"
          >
            <div
              class="square-container"
            >
              <div
                class="square most"
                data-testid="square-color"
              />
            </div>
          </div>
        </div>
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div
            class="tooltip-trigger"
            data-testid="tooltip-trigger"
          >
            <div
              class="square-container"
            >
              <div
                class="square nothing"
                data-testid="square-color"
              />
            </div>
          </div>
        </div>
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div
            class="tooltip-trigger"
            data-testid="tooltip-trigger"
          >
            <div
              class="square-container"
            >
              <div
                class="square most"
                data-testid="square-color"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="column"
      >
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div
            class="tooltip-trigger"
            data-testid="tooltip-trigger"
          >
            <div
              class="square-container"
            >
              <div
                class="square nothing"
                data-testid="square-color"
              />
            </div>
          </div>
        </div>
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div
            class="tooltip-trigger"
            data-testid="tooltip-trigger"
          >
            <div
              class="square-container"
            >
              <div
                class="square nothing"
                data-testid="square-color"
              />
            </div>
          </div>
        </div>
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div
            class="tooltip-trigger"
            data-testid="tooltip-trigger"
          >
            <div
              class="square-container"
            >
              <div
                class="square nothing"
                data-testid="square-color"
              />
            </div>
          </div>
        </div>
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div
            class="tooltip-trigger"
            data-testid="tooltip-trigger"
          >
            <div
              class="square-container"
            >
              <div
                class="square nothing"
                data-testid="square-color"
              />
            </div>
          </div>
        </div>
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div
            class="tooltip-trigger"
            data-testid="tooltip-trigger"
          >
            <div
              class="square-container"
            >
              <div
                class="square most"
                data-testid="square-color"
              />
            </div>
          </div>
        </div>
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div
            class="tooltip-trigger"
            data-testid="tooltip-trigger"
          >
            <div
              class="square-container"
            >
              <div
                class="square most"
                data-testid="square-color"
              />
            </div>
          </div>
        </div>
        <div
          class="tooltip"
          data-testid="tooltip"
        >
          <div
            class="tooltip-trigger"
            data-testid="tooltip-trigger"
          >
            <div
              class="square-container"
            >
              <div
                class="square nothing"
                data-testid="square-color"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
